scripts/ci/test_extensions.sh

```diff
@@ -36,11 +36,11 @@ for ext in $output; do
     fi
 done
 
-az self-test --debug
-if [ $? != 0 ]
-then
-    exit_code=1
-    echo "Failed to verify:" $ext
-fi
+#az self-test --debug
+#if [ $? != 0 ]
+#then
+#    exit_code=1
+#    echo "Failed to verify:" $ext
+#fi
```

Comment:
Comment `az self-test` as it cannot pass on ADO. The issue is caused by extensions `azure-iot` and `arcappliance` both have `jsonschema` dependency but with different versions. On ADO, `azure-iot` is now always installed before `arcappliance` (the install order is decided by `os.listdir(ext_dir)` which is related to the order of how files are stored in the file system).  Their directory are both appended to `sys.path`, and `azure-iot` is in front of `arcappliance`. Therefore, in `arcappliance`, when importing `jsonschema`, it will import from the one (version 3.0.2) installed in `azure-iot` and this line of code caused an import issue: `__version__ = get_distribution(__name__).version`. 

If on some system, `arcappliance` is installed before `azure-iot`, no error will be thrown as `arcappliance` uses `jsonschema` version 3.2.0 and the code to get version changes to `__version__ = metadata.version("jsonschema")`.


Actually, `az self-test` should not load all extensions together, `sys.path` will append all extension directories, and causing issues like above for the same dependency but with different versions in different extensions.

Reply:
> The issue is caused by extensions `azure-iot` and `arcappliance` both have `jsonschema` dependency but with different versions. 

Yes, `azure-iot/jsonschema` (3.0.2) overriding `arcappliance/jsonschema` (3.2.0) is an important contributing factor. I provided more detail at https://github.com/Azure/azure-cli/pull/17456

> Actually, `az self-test` should not load all extensions together

It is possible for CLI to load multiple extensions at the same time - when command index has not been built yet.